src/software/backend/wifi_backend.cpp

```diff
@@ -61,22 +61,24 @@ void WifiBackend::receiveRobotLogs(TbotsProto::RobotLog log)
 
 void WifiBackend::joinMulticastChannel(int channel, const std::string& interface)
 {
-    vision_output.reset(new ThreadedProtoMulticastSender<TbotsProto::Vision>(
-        std::string(MULTICAST_CHANNELS[channel]) + "%" + interface, VISION_PORT));
+    vision_output.reset(new ThreadedProtoUdpSender<TbotsProto::Vision>(
+        std::string(MULTICAST_CHANNELS[channel]) + "%" + interface, VISION_PORT, true));
 
-    primitive_output.reset(new ThreadedProtoMulticastSender<TbotsProto::PrimitiveSet>(
-        std::string(MULTICAST_CHANNELS[channel]) + "%" + interface, PRIMITIVE_PORT));
+    primitive_output.reset(new ThreadedProtoUdpSender<TbotsProto::PrimitiveSet>(
+        std::string(MULTICAST_CHANNELS[channel]) + "%" + interface, PRIMITIVE_PORT,
+        true));
 
-    robot_status_input.reset(new ThreadedProtoMulticastListener<TbotsProto::RobotStatus>(
+    robot_status_input.reset(new ThreadedProtoUdpListener<TbotsProto::RobotStatus>(
         std::string(MULTICAST_CHANNELS[channel]) + "%" + interface, ROBOT_STATUS_PORT,
         boost::bind(&Backend::receiveRobotStatus, this, _1)));
 
-    robot_log_input.reset(new ThreadedProtoMulticastListener<TbotsProto::RobotLog>(
+    robot_log_input.reset(new ThreadedProtoUdpListener<TbotsProto::RobotLog>(
         std::string(MULTICAST_CHANNELS[channel]) + "%" + interface, ROBOT_LOGS_PORT,
         boost::bind(&WifiBackend::receiveRobotLogs, this, _1)));
```

Comment:
these should all have the multicast flag set

Reply:
For listeners, I created an additional constructor instead of passing in a multicast flag where only a port is given without an ip address. This will create a listener that listens to local machine's any address for the given port.

Reply:
The interface should ideally be consistent w/ both the sender/listener, we have an extra constructor for the listener here w/ a flag for the sender

I think having the explicit multicast socket option flag (as you have in the sender) is better than silently joining a multicast group in a library called `ProtoUdpListener`, because its not immediately clear that it does anything related to multicast.

Having another constructor is fine, but it would be nice to still have the `multicast?` on the original constructor imo.

Thoughts? @jsn0716 @jonathanlew @kelvinkoon

Reply:
Sure, I can add the flag, so if `multicast` flag is false in that constructor, it would listen on a specific interface instead of any. Lmk if u have any other thoughts @jonathanlew @kelvinkoon

Reply:
if multicast is false, it wouldn't join the group on the socket. the rest can probably stay the same?

Reply:
Added multicast option to udp listener constructor